### PR TITLE
Datasource

### PR DESCRIPTION
**Pull request by Fraideron**
Created basic structures, for datasource usage

### CODE REVIEW

File: src/lib/Datasource.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Datasource {
```
Comment (vladyslav-dotsenko) [inline src/lib/Datasource.js:2]: Please, consider to add 'use strict' line at the start of every file.

File: src/lib/FileReadDataSource.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const fs = require('fs');
```
Comment (vladyslav-dotsenko) [inline src/lib/FileReadDataSource.js]: This file should be under `./datasource` folder

File: src/lib/FileReadWriteDataSource.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+const fs = require('fs');
```
Comment (vladyslav-dotsenko) [inline src/lib/FileReadWriteDataSource.js]: This file should be under `./datasource` folder

File: src/lib/InMemoryDataSource.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+const Datasource = require('../Datasource');
```
Comment (vladyslav-dotsenko) [inline src/lib/InMemoryDataSource.js]: This file should be under `./datasource` folder

File: test.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const InMemoryDataSource = require('./tmp/InMemoryDataSource');
```
Comment (vladyslav-dotsenko) [inline test.js]: Unneeded dev-testing file

File: test/FileReadDataSource.test.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
+const expect = require('expect');
```
Comment (vladyslav-dotsenko) [inline test/FileReadDataSource.test.js]: Adjust those tests to fit any abstract datasource

File: test/FileReadWriteDataSource.test.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+const expect = require('expect');
```
Comment (vladyslav-dotsenko) [inline test/FileReadWriteDataSource.test.js]: Adjust those tests to fit any abstract datasource

File: test/fixtures/fixtureDataCorrect.json [hunk 8]
```diff
@@ -0,0 +1 @@
+{"state":"Ukraine"}
```
Comment (vladyslav-dotsenko) [inline test/fixtures/fixtureDataCorrect.json]: No need to name your file as `fixtureBlaBla` if it us under `fixtures` folder.
The only good naming convention with words repeat is `./fixtures/filename.fixture.json` but think it is redundant as well.

**Review by vladyslav-dotsenko — CHANGES_REQUESTED** [review]
Please, fix all the minors of naming and conventions before merge and remove redundant things.

File: src/lib/Datasource.js (hunk 1 above)
Comment (Fraideron) [reply src/lib/Datasource.js:2]: done

File: src/lib/FileReadDataSource.js (hunk 2 above)
Comment (Fraideron) [reply src/lib/FileReadDataSource.js]: done

File: src/lib/FileReadWriteDataSource.js (hunk 3 above)
Comment (Fraideron) [reply src/lib/FileReadWriteDataSource.js]: done

File: src/lib/InMemoryDataSource.js (hunk 4 above)
Comment (Fraideron) [reply src/lib/InMemoryDataSource.js]: done

File: test.js (hunk 5 above)
Comment (Fraideron) [reply test.js]: removed

File: test/fixtures/fixtureDataCorrect.json (hunk 8 above)
Comment (Fraideron) [reply test/fixtures/fixtureDataCorrect.json]: done

File: test/FileReadDataSource.test.js (hunk 6 above)
Comment (Fraideron) [reply test/FileReadDataSource.test.js]: Created a responsive testing file with unic methods for all datasources. Needed some review

**Review by vladyslav-dotsenko — APPROVED** [review]

**Pull request merged**